parser.py

```diff
@@ -513,11 +512,17 @@ def parse_direct_messages(data_folder, users, user_id_URL_template, output_dms_f
             messages.sort(key=lambda tup: tup[0])
             markdown += ''.join(md for _, md in messages)
             num_written_messages += len(messages)
-        dms_markdown += '\n\n----\n\n' + markdown
-    # output as a single file for now
-    with open(output_dms_filename, 'w', encoding='utf8') as f:
-        f.write(dms_markdown)
-    print(f"Wrote {len(dms_json)} conversations ({num_written_messages} total messages) to {output_dms_filename}")
+
+            # output as one file per conversation
+            other_user_id = user2_id if user1_handle == username else user1_id
+            other_user: str = users[other_user_id].handle if other_user_id in users else other_user_id
+            conversation_output_filename = f'DMs-Archive-{other_user}.md'
```

Comment:
Please move outside the function as a template `conversation_output_filename_template` to where `output_dms_filename` was.